FuelSDK-CSharp/ETClient.cs

```diff
@@ -91,8 +94,12 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
                 {
                     configSection.RestEndPoint = parameters["restEndPoint"];
                 }
+                if (parameters.AllKeys.Contains("oAuth2AuthenticationMode"))
+                {
+                    configSection.RestEndPoint = parameters["oAuth2AuthenticationMode"];
+                }
             }
-
+            UseOAuth2Authentication = String.IsNullOrEmpty(configSection.OAuth2AuthenticationMode) ? false : Boolean.Parse(configSection.OAuth2AuthenticationMode);
```

Comment:
Instead of using the ternary operator here, the logic to determine if the `OAuth2AuthenticationMode` is ON could be moved in the `FueldSDKConfigurationSection.OAuth2AuthenticationMode` property getter. 

Also, `UseOAuth2Authentication` local variable could be removed and `configSection.OAuth2AuthenticationMode` can be used instead. This mean that the `Binding GetSoapBinding(bool useOauth2 = false)` method can be simplified to drop the parameter and use `configSection.OAuth2AuthenticationMode` inside its body.

Reply:
Moved the logic to getter. But we still need the Property in ETClient as it is used outside of ETClient. And ConfigSection is private. I didn't want to touch anything existing.